tests/net/hostname/src/main.c

```diff
@@ -90,6 +101,17 @@ static void net_iface_init(struct net_if *iface)
 			     NET_LINK_ETHERNET);
 }
 
+#if CONFIG_NET_MGMT_EVENT
+static void hostname_changed(struct net_mgmt_event_callback *cb,
+			     uint32_t mgmt_event, struct net_if *iface)
+{
+	if (mgmt_event == NET_EVENT_IF_HOSTNAME_CHANGED) {
```

Comment:
Check also that the info is properly set. For example that `net_hostname_get()` returns the same data as `cb->info`

Reply:
done